packages/ckeditor5-table/src/tablekeyboard.js

```diff
@@ -51,10 +51,18 @@ export default class TableKeyboard extends Plugin {
 		this.editor.keystrokes.set( 'Tab', this._getTabHandler( true ), { priority: 'low' } );
 		this.editor.keystrokes.set( 'Shift+Tab', this._getTabHandler( false ), { priority: 'low' } );
 
+		const priorityAboveWidget = priorities.get( 'high' ) + 10;
+		const priorityBelowWidget = priorities.get( 'high' ) - 10;
+
+		// This listener is handling cases inside table cell.
 		// Note: This listener has the "high-10" priority because it should allow the Widget plugin to handle the default
 		// behavior first ("high") but it should not be "prevent–defaulted" by the Widget plugin ("high-20") because of
 		// the fake selection retention on the fully selected widget.
-		this.listenTo( viewDocument, 'keydown', ( ...args ) => this._onKeydown( ...args ), { priority: priorities.get( 'high' ) - 10 } );
+		this.listenTo( viewDocument, 'keydown', ( ...args ) => this._onKeydown( false, ...args ), { priority: priorityBelowWidget } );
```

Comment:
Boolean handling in this class is getting out of control. Basically every method has `true` / `false` in parameters and this is getting hard to read.

I don't know what this action does `onKeydown( false, ...args )` is very cryptic. If there are two distinguish actions then it should be reflected in the method name.

I would see here two methods:

```javascript
this.listenTo( viewDocument, 'keydown', ( ...args ) => this._handleKeydownOutsideTable( ...args ), ... );
this.listenTo( viewDocument, 'keydown', ( ...args ) => this._handleKeydownInsideTable( ...args ), ... );
```

Alternative might be a method that would return handler:

```javascript
this.listenTo( viewDocument, 'keydown', this._getKeydownOutsideTableHandler(), { ... } );
this.listenTo( viewDocument, 'keydown', this._getKeydownInsideTableHandler(), { ... } );
```

Both give more information about the intention of the method.

The other problem is that `highPriority` as a method parameter does not tell what this flag do.

Reply:
The other solution could be string - less refactoring:

```javascript
this.listenTo( viewDocument, 'keydown', this._getKeyDownHandler( 'outside' ), { ... } );
this.listenTo( viewDocument, 'keydown', this._getKeyDownHandler( 'inside' ), { ... } );
```

But still, I found separate handlers better in terms of separation of concerns here.

Reply:
Done :)